userspace/libscap/scap.h

```diff
@@ -1055,6 +1073,27 @@ int32_t scap_suppress_events_comm(scap_t* handle, const char *comm);
 
 bool scap_check_suppressed_tid(scap_t *handle, int64_t tid);
 
+/*!
+  \brief Create an event from the parameters given as arguments.
+
+  Create any event from the event_table passing the type, n and the parameters as variadic arguments as follows:
+   - Any integer type is passed from the correct type
+   - String types (including PT_FSPATH, PT_FSRELPATH) are passed via a null-terminated char*
+   - Buffer types, variable size types and similar, including PT_BYTEBUF, PT_SOCKTUPLE are passed with
+     a struct scap_sized_buffer
+
+  \param event_buf A pointer to a scap_sized_buffer. If the contained buf is NULL or the length is 0 it will be allocated.
+  If the buffer is too small to contain the event it will be reallocated. Pointer and size are updated.
+  Even in case of error, this function may allocate a valid buffer. The caller must manage its lifecycle.
+  \param error A pointer to a scap error string to be filled in case of error.
+  \param event_type The event type (normally PPME_*)
+  \param n The number of parameters for this event.
+  This is required as the number of parameters used for each event can change between versions.
+  \param ...
+  \return int32_t The error value
+ */
+int32_t scap_event_encode_params(struct scap_sized_buffer *event_buf, char *error, enum ppm_event_type event_type, uint32_t n, ...);
```

Comment:
Same thing here about using iovecs or at least allowing this function to populate one of an array of buffers.